model/user/db.go

```diff
@@ -414,6 +414,25 @@ func ClearLoginCache(user gimlet.User) error {
 	return nil
 }
 
+// ClearUserSettings clears one user's settings
+func ClearUserSettings(userId string) error {
+	update := bson.M{"$unset": bson.M{SettingsKey: 1}}
```

Comment:
Rather than unset, can we `$set` it to an empty doc? I worry that we've set the BSON tag to not specify `omitempty`, so we should be able to assume for DB model consistency that the settings subdoc is either populated or empty (but not null/unset).

Reply:
done. but since it doesn't have `omitempty` set, the end result is still the same, the db will still hold the struct with the values unset.

Reply:
The distinction I was making is that `$set`'ing it to the empty doc will result in the doc looking like:
```
{
    "settings": {}
}
```
which is consistent with the way the data model is defined. In contrast, `$unset` will actually delete the field completely, so that field will no longer exist in the doc at all.

Reply:
However, I had a unit test set up for this function and in both cases (before and after the change) it went form a settings struct with fields, to 

` UserSettings{
		Timezone: "", Region: "", GithubUser: GithubUser{UID: 0, LastKnownAs: ""},
		SlackUsername: "", Notifications: NotificationPreferences{BuildBreak: "",
			BuildBreakID: "", PatchFinish: "", PatchFinishID: "", PatchFirstFailure: "",
			PatchFirstFailureID: "", SpawnHostExpiration: "", SpawnHostExpirationID: "",
			SpawnHostOutcome: "", SpawnHostOutcomeID: "", CommitQueue: "", CommitQueueID: ""},
		UseSpruceOptions: UseSpruceOptions{PatchPage: false, SpruceV1: false, HasUsedSpruceBefore: false, HasUsedMainlineCommitsBefore: false}`

Reply:
Yeah, the Golang struct will look like this because this is the default zero value for the struct, but the actual DB doc won't have the field set at all.